js/apps/admin-ui/src/identity-providers/add/AdvancedSettings.tsx

```diff
@@ -125,6 +136,92 @@ export const AdvancedSettings = ({ isOIDC, isSAML }: AdvancedSettingsProps) => {
 
+      {(!isSAML || isOIDC) && (
+        <FormGroupField label="filteredByClaim">
+          <Controller
+            name="config.filteredByClaim"
+            defaultValue="false"
+            control={control}
+            render={({ field }) => (
+              <Switch
+                id="filteredByClaim"
+                label={t("common:on")}
+                labelOff={t("common:off")}
+                isChecked={field.value === 'true'}
+                onChange={(value) => {
+                    const v=""+value
+                    field.onChange(v)
+                    setClaimFilterRequired(value)
+                  }
+                }
+                aria-label="filteredByClaim"
+              />
+            )}
+          />
+        </FormGroupField>
+      )}
+      {(!isSAML || isOIDC) && claimFilterRequired && (
+        <>
+          <FormGroup
+            label={t("identity-providers:claimFilterName")}
+            labelIcon={
+              <HelpItem
+                helpText={t("identity-providers-help:claimFilterName")}
+                fieldLabelId="identity-providers:claimFilterName"
+              />
+            }
+            fieldId="kc-claim-filter-name"
+            isRequired
+            validated={
+              errors.config?.claimFilterName
+                ? ValidatedOptions.error
+                : ValidatedOptions.default
+            }
+            helperTextInvalid={t("common:required")}
+          >
+            <KeycloakTextInput
+              isRequired
+              id="kc-claim-filter-name"
+              data-testid="claimFilterName"
+              validated={
+                errors.config?.claimFilterName
+                ? ValidatedOptions.error
+                : ValidatedOptions.default
+              }
+              {...register("config.claimFilterName", { required: true })}
+            />
+          </FormGroup>
+          <FormGroup
+            label={t("identity-providers:claimFilterValue")}
+            labelIcon={
+              <HelpItem
+                helpText={t("identity-providers-help:claimFilterValue")}
+                fieldLabelId="identity-providers:claimFilterName"
+              />
+            }
+            fieldId="kc-claim-filter-value"
+            isRequired
+            validated={
+              errors.config?.claimFilterValue
+                ? ValidatedOptions.error
+                : ValidatedOptions.default
+            }
+            helperTextInvalid={t("common:required")}
+          >
+            <KeycloakTextInput
+              isRequired
+              id="kc-claim-filter-value"
+              data-testid="claimFilterValue"
+              validated={
+                errors.config?.claimFilterValue
+                ? ValidatedOptions.error
+                : ValidatedOptions.default
+              }
+              {...register("config.claimFilterValue", { required: claimFilterRequired })}
```

Comment:
I am not sure you can update the `required` value here after it has been passed. It might have to be set explicitly with a different API.

Have you tested that this works as expected?